### ISSUE TITLE
Multiline field is broken

### ISSUE THREAD

**Issue by escopecz** [magento/magento2#18115]
### Preconditions

Magento 2.2.4
API authorization OAuth1a

### Steps to reproduce

1. Create new multiline custom field on the customer object.
2. Fill in some values for a customer like this:
![screen shot 2018-09-18 at 13 56 38](https://user-images.githubusercontent.com/1235442/45685788-d88ea080-bb4a-11e8-9c7a-7fbca08ca1d6.png)
3. Save the customer.
4. Open the edit form for that customer again. All the values are gone like this:
![screen shot 2018-09-18 at 13 57 18](https://user-images.githubusercontent.com/1235442/45685824-ee9c6100-bb4a-11e8-85d0-459d7a70ade2.png)
5. Let's see what the API will return us for that field:
```
  "custom_attributes": [
    {
      "attribute_code": "multiline",
      "value": "aaa"
    }
  ]
```
Hmm, that is interesting... It's the value of the first field as a string. So multiline fields will return string as a value? Wait, the **Street** field is also type of multiline. What that field returns in the API?
```
    "street": [
        "Address 11s5"
      ],
```
So this multi line field can return a string and/or an array? You know what's interesting too? The street field require an array when you want to save a value to that field via API. But another multiline field requires a string. What is the pattern here?

### Expected result

1. Multiline field will store the values when the customer is saved.
2. Multiline field will return and expect consistent value type via API.

### Actual result

1. Multiline field does not store the values when the customer is saved.
2. Multiline field does not return and expect consistent value type via API.

**Comment by magento-engcom-team** [comment]
Hi @escopecz. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@escopecz do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `G1 Passed` will be added to the issue automatically. Please, edit issue description if needed, until label `G1 Passed` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `G2 Passed` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `acknowledged` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report is acknowledged.

**Comment by ghost** [comment]
Hi @escopecz, thank you for your report. Customer attribute available only on EE version. Please follow these guidelines for proper tracking of your issue. You can report Commerce-related issues in one of two ways:
You can use the Support portal associated with your account
or
If you are a Partner reporting on behalf of a merchant, use the Partner portal.
 
GitHub is intended for Magento Open Source users to report on issues related to Open Source only. There are no account management services associated with GitHub.

**Comment by ghost** [comment]
@escopecz also duplicate for this -> https://github.com/magento/magento2/issues/8086

**Comment by escopecz** [comment]
I saw that ticket. I find my ticket more descriptive and actually all I care about is the API problem that it returns/expects sometimes array and sometimes string. This is not mentioned at the other ticket at all. Please reopen.

**Comment by ghost** [comment]
@escopecz No, problem described in referenced ticked actually the same, custom attribute not save correctly, so in the API you have the same result, if someone fixed referenced ticket, you resolve you issue.
And referenced issue already have created ticket, and someone maybe work for this

**Comment by escopecz** [comment]
How come the street multiline field works and other don't? I have to implement the API communication for this field and it's nightmare as I don't know what the rule for the field value type is. I'd remove this type from field mapping completely as it's buggy, but the street field is important.

**Comment by ghost** [comment]
@escopecz 
Street Address use another logic of attributes and this works for magento core, but for custom attributes  not.
As i can see this bug is acknowledged, and we are waiting for maintainer to fix this.
Also you can use multiselect attribute.

**Comment by escopecz** [comment]
Thanks a lot for explanation! That's what I needed. So I'll ignore all multiline fields that are not system.

**Comment by amenk** [comment]
Did anybody here already work on making Magento allowing to change a field type (company in our case) to multiline and make it work across the frontend and API? We are wondering if it is better to implement this in the core and provide a pull request or make a own custom module.

Any feedback is appreciated.

*Sorry for the forum-like post*

**Comment by vivekkumarcedcoss** [comment]
@amenk & @escopecz you can review the PR #20371 , this issue was fixed in that PR.

**Comment by amenk** [comment]
@vivekkumarcedcoss but that is only for Admin Config inputs, right?

**Comment by vivekkumarcedcoss** [comment]
Yes @amenk that PR is only for Configuration.

**Comment by magento-engcom-team** [comment]
Hi @escopecz. Thank you for your report. 
The issue has been fixed in magento/magento2#20371 by @vivekkumarcedcoss in 2.3-develop branch
Related commit(s):
 - [ddb1137b96ff0ae93c2e78dff62e02da31f9a02a](https://github.com/magento/magento2/commit/ddb1137b96ff0ae93c2e78dff62e02da31f9a02a)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @escopecz. Thank you for your report. 
The issue has been fixed in magento/magento2#21561 by @vivekkumarcedcoss in 2.2-develop branch
Related commit(s):
 - [96c83295a26dd07e1819e310d2d31fef25fadec9](https://github.com/magento/magento2/commit/96c83295a26dd07e1819e310d2d31fef25fadec9)

The fix will be available with the upcoming 2.2.9 release.
